### PR TITLE
fix(hash_with_args): honor explicit default_seed=0 instead of env fal…

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
## Summary

The HashWithArgs ETL transformer (all three server variants: FastAPI, Flask, HTTP multi-threaded) silently ignored an explicitly passed `default_seed=0` and fell back to the `SEED_DEFAULT` environment variable instead. Any caller constructing `HashWithArgs(default_seed=0)` with `SEED_DEFAULT` set to a non-zero value would compute hashes with the wrong seed, producing digests that do not match the caller's expectation.

## Root cause

```python
if default_seed:
    self.default_seed = default_seed
else:
    self.default_seed = int(os.getenv("SEED_DEFAULT", "0"))
```

The truthiness check treats `0` (a perfectly valid xxhash seed) as "not provided", so the env fallback path runs even when the caller explicitly requested seed 0. The docstring states the env var is read only "If None".

## Fix

Change the truthiness check to an explicit `None` check in all three server variants:

```python
if default_seed is not None:
```

## Testing

Repo tests (`transformers/tests/test_hash_with_args.py`) are integration tests requiring a running AIStore cluster and cannot run locally. Per review feedback, the integration test now explicitly transforms each file with seed 0 (`etl_args="0"`) and asserts the digest matches `xxh64(seed=0)`, covering the falsy-zero edge case end-to-end.

Smoke test (`uv run --with xxhash python smoke.py`, stubbing the aistore SDK import) with `SEED_DEFAULT=42`:

- `HashWithArgs(default_seed=0).default_seed == 0` — PASS (was 42 pre-fix)
- `HashWithArgs().default_seed == 42` (env fallback) — PASS
- `HashWithArgs(default_seed=7).default_seed == 7` — PASS
- `transform(b"hello", ...)` matches `xxhash.xxh64(seed=0)` — PASS

With the fix stashed (unmodified code), the first assertion fails with `default_seed=0 gave 42`, confirming the bug and that the fix resolves it.

## Why existing tests missed it

The existing test only exercised seeds via `etl_args` (random ints in [0, 1000] passed per-request) and never pinned the seed-0 edge case, so the falsy-`0` path was uncovered. The integration test now covers seed 0 explicitly.

### CODE REVIEW

File: transformers/tests/test_hash_with_args_unit.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
```
Comment (gaikwadabhishek) [inline transformers/tests/test_hash_with_args_unit.py]: remove this, and test part of integration tests. only add test for 0

**Review by gaikwadabhishek — CHANGES_REQUESTED** [review]
LGTM, rebase and squash

**Comment by andrewwhitecdw** [comment]
Feedback addressed — thanks @gaikwadabhishek:

- Removed `transformers/tests/test_hash_with_args_unit.py`; the seed-0 case now lives in the integration test (`_verify_test_files` in `test_hash_with_args.py` explicitly transforms with `args="0"` and asserts `xxh64(seed=0)`).
- Rebased onto latest `main` and squashed to a single commit (`fa94af3`).

Ready for another look.

**Comment by gaikwadabhishek** [comment]
LGTM, can you please add a sign off the commit and add the commit desc in small concise bullets instead of a blob desc. Just following the parent repo's convention

**Comment by andrewwhitecdw** [comment]
Done — amended the commit to include `Signed-off-by: Andrew White <andrewh@cdw.com>` and rewrote the body as concise bullets instead of the paragraph form.

**Review by gaikwadabhishek — APPROVED** [review]
